app/src/composables/use-item.ts

```diff
@@ -37,11 +37,11 @@ type UsableItem = {
 	validationErrors: Ref<any[]>;
 };
 
-export function useItem(
+export function useItem<T extends Record<string, any>>(
 	collection: Ref<string>,
 	primaryKey: Ref<string | number | null>,
 	query: Ref<Query> | Query = {}
-): UsableItem {
+): UsableItem<T> {
 	const { info: collectionInfo, primaryKeyField } = useCollection(collection);
 	const item = ref<Record<string, any> | null>(null);
```

Comment:
To fix the issue marked todo
> // TODO Find way to fix warning:
> // 'T' could be instantiated with a different subtype of constraint 'Record<string, any>'

```suggestion
	const item: Ref<T | null> = ref(null);
```

Reply:
Thanks! I think I've tried this as well but noticed that it moves the issue further down:
<img width="701" src="https://github.com/directus/directus/assets/5363448/0b03b058-bdd0-4f88-be7e-e280b724ad2a">

I've now added an assertion which should be fine in this case 👍 
https://github.com/directus/directus/blob/ed859ef8e8529bd7b0c56cf2deaf9f741332b538/app/src/composables/use-item.ts#L400